src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -484,6 +493,27 @@ protected async Task<ConnectNewHeadersResult> PresentHeadersLockedAsync(List<Blo
             return result;
         }
 
+        private bool CheckIfUnsolicitedFutureHeader(List<BlockHeader> headers)
```

Comment:
@zeptin so nothing in this bit of logic changed? It just got moved so that it can be re-used?

Reply:
Yes, that is the intention